crates/chain/src/tx_graph.rs

```diff
@@ -937,9 +1006,14 @@ impl<A: Anchor> TxGraph<A> {
 
     /// List txids by descending last-seen order.
     ///
-    /// Transactions without last-seens are excluded.
-    pub fn txids_by_descending_last_seen(&self) -> impl ExactSizeIterator<Item = (u64, Txid)> + '_ {
-        self.txs_by_last_seen.iter().copied().rev()
+    /// Transactions without last-seens are excluded. Transactions with a last-evicted timestamp
+    /// equal or higher than it's last-seen timestamp are excluded.
+    pub fn txids_by_descending_last_seen(&self) -> impl Iterator<Item = (u64, Txid)> + '_ {
+        self.txs_by_last_seen
+            .iter()
+            .copied()
+            .rev()
+            .filter(|(last_seen, txid)| !matches!(self.last_evicted.get(txid), Some(last_evicted) if last_evicted >= last_seen))
```

Comment:
Only if you consider this is better than `!matches!(...)`
```suggestion
           .filter(|(last_seen, txid)| {
               self.last_evicted
                   .get(txid)
                   // Has the current tx been seen after its last eviction?
                   .map_or(true, |last_evicted| last_evicted < last_seen)
           })
```